src/layers/basic.jl

```diff
@@ -253,3 +253,79 @@ end
 function Base.show(io::IO, b::SkipConnection)
   print(io, "SkipConnection(", b.layers, ", ", b.connection, ")")
 end
+
+"""
+    Bilinear(in1, in2, out)
+
+Creates a Bilinear layer, which operates on two inputs at the same time.
+It has parameters `W` and `b`, and its output given vectors `x`, `y` is of the form 
+
+    z[i] = σ.(x' * W[i,:,:] * y .+ b[i])
```

Comment:
If I'm reading this correctly, the dots here don't do anything:
```suggestion
It has parameters `W` and `b`, and its output given vectors `x`, `y` is another vector `z` with 

    z[i] = σ(x' * W[i,:,:] * y + b[i])
```
It could also be written `dot(x, W[i,:,:], y) + b[i]` but I don't know if that's clearer.